controller/deployment.go

```diff
@@ -30,19 +30,34 @@ func NewDeploymentRepo(db *postgres.DB, pgxpool *pgx.ConnPool) *DeploymentRepo {
 }
 
 func (r *DeploymentRepo) Add(data interface{}) error {
-	deployment := data.(*ct.Deployment)
-	if deployment.ID == "" {
-		deployment.ID = random.UUID()
+	d := data.(*ct.Deployment)
+	if d.ID == "" {
+		d.ID = random.UUID()
 	}
-	query := "INSERT INTO deployments (deployment_id, app_id, old_release_id, new_release_id, strategy) VALUES ($1, $2, $3, $4, $5) RETURNING created_at"
-	if err := r.db.QueryRow(query, deployment.ID, deployment.AppID, deployment.OldReleaseID, deployment.NewReleaseID, deployment.Strategy).Scan(&deployment.CreatedAt); err != nil {
-		return err
+	if d.OldReleaseID == "" {
+		query := "INSERT INTO deployments (deployment_id, app_id, new_release_id, strategy) VALUES ($1, $2, $3, $4) RETURNING created_at"
+		if err := r.db.QueryRow(query, d.ID, d.AppID, d.NewReleaseID, d.Strategy).Scan(&d.CreatedAt); err != nil {
+			return err
+		}
+	} else {
+		query := "INSERT INTO deployments (deployment_id, app_id, old_release_id, new_release_id, strategy) VALUES ($1, $2, $3, $4, $5) RETURNING created_at"
```

Comment:
You can avoid this by passing a **string which will insert NULL if the inner pointer is nil. We do this for artifact_ids on releases.